src/Portable.Xaml/Portable.Xaml.ComponentModel/DateTimeConverter.cs

```diff
@@ -45,8 +58,28 @@ public override object ConvertTo (ITypeDescriptorContext context, CultureInfo cu
 
                 return date.ToString(format, culture);
             }
-			return base.ConvertTo (context, culture, value, destinationType);
-		}
-	}
+            return base.ConvertTo(context, culture, value, destinationType);
+        }
+
+        private string RemoveUnnecessaryZeros(string dateFormatted, DateTimeKind kind)
+        {
+            string conciseDate = dateFormatted.Trim();
+            if (kind == DateTimeKind.Utc)
+            {
+                if (UtcTidyUpPattern.IsMatch(dateFormatted))
+                    conciseDate = UtcTidyUpPattern.Replace(dateFormatted, "Z");
+                if (conciseDate.EndsWith(".Z"))
+                    conciseDate = conciseDate.Replace(".Z", "Z");
+            }
+            else
+            {
+                if (LocalTidyUpPatterns.IsMatch(dateFormatted))
+                    conciseDate = LocalTidyUpPatterns.Replace(dateFormatted, "");
+                if (conciseDate.EndsWith("."))
+                    conciseDate = conciseDate.Replace(".", string.Empty);
```

Comment:
This should be `conciseDate = conciseDate.Substring(0, consiceDate.Length - 1)` for efficiency and to ensure it doesn't replace any other dots in the string.